Frontend/src/Components/LogIn.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAuth } from './AuthContext';
+import './Login.css';
+
+const backendApi = import.meta.env.VITE_BACKEND_ADDRESS
+
+const LogIn = () => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const { login } = useAuth();
+  const navigate = useNavigate();
+
+  const handleLogin = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await fetch(`${backendApi}/login`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+       
+        body: JSON.stringify({ username, password }),
+      });
+
+      if (response.ok) {
+        login();
+        navigate('/'); // Redirect to home page
+      } else {
+        setError('Invalid credentials. Please try again.');
+      }
+    } catch (err) {
+      setError('Something went wrong. Please try again.');
+    }
+  };
+
+  return (
+    <div className="loginContainer">
+      <header className="loginHeader">
+        <div className="appName">World Collection</div>
+        <div className="authLinks">
+          <a href="/login">Log in</a>
+          <a href="/register">Sign up</a>
```

Comment:
Usually it's a good idea to use `Link` tag provided by react router instead of `a` tag, as `a` tag takes relative path and `Link` tag is absolute.

Reply:
And `a` tag will trigger a full page refresh from what I know, and we might want to avoid that.